avm/res/compute/disk-encryption-set/modules/nested_keyVaultPermissions.bicep

```diff
@@ -0,0 +1,66 @@
+@description('Required. A boolean to specify whether or not the used Key Vault has RBAC authentication enabled or not.')
+param rbacAuthorizationEnabled bool = true
+
+@description('Required. The resourceID of the User Assigned Identity to assign permissions to.')
+param userAssignedIdentityResourceId string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Required. Resource ID of the KeyVault containing the key or secret.')
+param keyVaultResourceId string
+
+@description('Required. Key URL (with version) pointing to a key or secret in KeyVault.')
+param keyName string
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: last(split(keyVaultResourceId, '/'))!
+
+  resource key 'keys@2021-10-01' existing = {
+    name: keyName
+  }
+}
+
+resource userAssignedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = {
+  name: last(split(userAssignedIdentityResourceId, '/'))!
+  scope: resourceGroup(split(userAssignedIdentityResourceId, '/')[2], split(userAssignedIdentityResourceId, '/')[4])
+
+}
+
+// =============== //
+// Role Assignment //
+// =============== //
+
+resource keyVaultKeyRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (rbacAuthorizationEnabled == true) {
+  name: guid('msi-${keyVault::key.id}-${location}-${userAssignedIdentityResourceId}-Key-Reader-RoleAssignment')
+  scope: keyVault::key
+  properties: {
+    principalId: userAssignedIdentity.properties.principalId
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '12338af0-0e69-4776-bea7-57ae8d297424') // Key Vault Crypto User
+    principalType: 'ServicePrincipal'
+  }
+}
+
+// ============= //
+// Access Policy //
+// ============= //
+
+module keyVaultAccessPolicies '../../../key-vault/vault/access-policy/main.bicep' = if (rbacAuthorizationEnabled != true) {
```

Comment:
@AlexanderSehr should we actually reference this module from the public registry?

Reply:
No, not at this time. While we fortunately kept the door open for that possibility, it's not yet a priority to publish child-modules.

That said, you'd need to do what @rahalan did in the VM PR where he referenced the protected item child module of the RSV module. That is, to add a copy of the accessPolicy `main.bicep` as e.g., a `key-vault.vault.access-policies.bicep` (or whatever you want to call it) to a subfolder `modules` in the `disk-encryption-set` module. So that would look like:

- disk-encryption-set
  - modules
    - `key-vault.vault.access-policies.bicep` (contains the copied code) 
  - `main.bicep` (references `key-vault.vault.access-policies.bicep`)

It's the same option contributors have for, for example, the Private Endpoint module. You don't 'have to' reference the published module but could instead have your own copy if you want.

Reply:
done: https://github.com/Azure/bicep-registry-modules/pull/749/commits/d1f1ae4e43de6ae80e82c001123fbfe7df8a7c5d